### ISSUE TITLE
[Bug]: ChoiceGroup focus conflict when it's the first component inside a focus zone

### ISSUE THREAD

**Issue by liyatilahun** [microsoft/fluentui#25058]
### Library

React / v8 (@fluentui/react)

### System Info

```shell
N/A
```

### Are you reporting Accessibility issue?

yes

### Reproduction

 https://codesandbox.io/s/focustrapzone-choicegroup-focus-conflict-9d96j?file=/src/index.tsx:79-92 (Comment out the TextField inside the code to repro the issue)

## Actual Behavior
The focus on the radio button does not focus on the selected option, but instead focus on the first option of the choice group. This happens when using keyboard navigation ( keyUp and keyDown and space). But it works as expected for shift-tab and tab. To repro this problem the ChoiceGroup should be the first component inside the focus zone.

## Expected Behavior
The focus on the radio button should focus on the selected option instead of the first option when using keyboard navigation.

### Requested priority

Normal

### Products/sites affected

word-online, ppt-online

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by SavannahBourgeois** [comment]
@smhigley as FYI since this is an accessibility bug we are seeing in some of our dialogs where the focus rect is not visible initially on a choicegroup

**Comment by TristanWatanabe** [comment]
@liyatilahun I seem to be unable to repro the issue with the latest v8. Can you please confirm if this has been fixed? Or if I'm missing something as far as how to repro the issue in which case, please let me know!

![ChoiceGroup Focus](https://user-images.githubusercontent.com/8649804/213513054-0802b829-5dbd-45f4-a36a-d44ef8db26b2.gif)

**Comment by liyatilahun** [comment]
Hi @TristanWatanabe, for the key down, I wasn't able to repro the issue. 

For key up, I was able to repro the issue. When I press key up, the focus goes to Option D. I was expecting Option A since that is the choice above option B.
![choice_group_key_up_bug](https://user-images.githubusercontent.com/92185632/213592490-63310397-32ab-404b-8744-2cf9e5fe943c.gif)

For space key I'm still seeing the issue. When I press on Space key, the focus goes to the first option and not the selected option.
![choice_group_space_key_bug](https://user-images.githubusercontent.com/92185632/213593693-1aab75d0-bd94-4906-9b12-6022f4c926dc.gif)

**Comment by TristanWatanabe** [comment]
Ok I was finally able to repro this - the steps taken were:
1) Open control in a separate window: https://h7hp77.csb.app/
2) Tab into `ChoiceGroup` (Option B will be focused)
3) Click anywhere outside of control
4) Observe as focus indicator goes to Option A
5) Pressing KeyUp will move focus to Option D instead of Option A

The root of the issue seems to be that when navigating away from the control, focus always goes back to the first option.
